src/app/action-items/action-items.component.ts

```diff
@@ -49,26 +57,30 @@ export class ActionItemsComponent implements OnInit, OnDestroy {
       .subscribe(tasks => {
         this.dataSource = tasks;
         this.isLoadingActionItems = false;
-        this.isLoadingActionItem = false;
+        this.showImageWhenNoActionItem = false;
+        if (tasks.length === 0) this.showImageWhenNoActionItem = true;
       });
   }
   openDialog(): void {
     const dialogConfig = new MatDialogConfig();
-    dialogConfig.data = { width: '470px', height: 'auto', disableClose: true };
+    dialogConfig.data = {
+      ...this.dialogData,
+      data: {}
+    };
     this.matDialog
-      .open(AddItemComponent, dialogConfig.data)
+      .open(AddOrUpdateActionItemComponent, dialogConfig.data)
       .afterClosed()
       .subscribe(item => {
         if (!!item) {
           this.retrieveActionItems();
-          this.justAddedItemId = item.id;
-          this.router.navigate(['/items'], { queryParams: { id: this.justAddedItemId } });
+          this.actionItemIdForScroll = item.id;
+          this.router.navigate(['/items'], { queryParams: { id: this.actionItemIdForScroll } });
         }
       });
   }
-  private triggerScrollTo() {
+  private triggerScrollTo(itemId: string): void {
```

Comment:
**CONCERN:** Shouldn't we clear value of `this.actionItemIdForScroll` after scrolling?